File: solutions/q1768/q1768.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.length();
        int m = word2.length();

        int i = 0;
        int j = 0;

        string ans = "";

        while (i < n && j < m) {
            ans += word1[i];
            ans += word2[j];
            i++;
            j++;
        }
        while (i < n) {
            ans += word1[i];
            i++;
        }
        while (j < m) {
            ans += word2[j];
            j++;
        }
        return ans;
    
    }
};

File: leetcode/q1768/q1768.kt
class Solution {
    fun mergeAlternately(word1: String, word2: String): String {
        var result = CharArray(word1.length+word2.length) { ' ' }
        var index = 0
        var resultIndex = 0
        while (index < maxOf(word1.length,word2.length)) {
            if (index < word1.length){
                  result[resultIndex] = word1[index]
                  resultIndex++
                }
            if (index < word2.length){
                result[resultIndex] = word2[index]
                resultIndex++
            }
            
            index++  
        }
        return String(result)
    }
}
